File: Environments/Chess/ChessTesting/ChessTesting/Benchmarking/Exceptions/InvalidResourceReferenceException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Benchmarking.Exceptions
{
    internal class InvalidResourceReferenceException : Exception
    {
        public string Resource { get; }
        public string Reference { get; }

        public InvalidResourceReferenceException(string resource, string reference)
            : base($"Resource {resource} referenced from {reference} not found.")
        {
            Resource = resource;
            Reference = reference;
        }
    }
}

File: Environments/MicroRTS/MicroRTS-master/src/benchmarking/InvalidResourceReferenceException.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package benchmarking;

public class InvalidResourceReferenceException extends Exception
{
    public final String Resource;
    public final String Reference;

    public InvalidResourceReferenceException(String resource, String reference)
    {
        super("Resource {" + resource + "} referenced from {" + reference + "} not found.");
        Resource = resource;
        Reference = reference;
    }
}
